src/features/snap/components/ReportSnap.tsx

```diff
@@ -1,21 +1,53 @@
+import { t } from '@lingui/macro';
 import { useState, type FunctionComponent } from 'react';
+import { useAccount } from 'wagmi';
 
 import { ReportSnapModal } from './modals/ReportSnapModal';
 import { ReportButton } from '../../../components';
+import { useVerifiableCredential } from '../../../hooks';
+import { useSignErrorHandler } from '../../../hooks/useSignErrorHandler';
+import useToastMsg from '../../../hooks/useToastMsg';
 
 type SnapReportProps = {
+  snapId: string;
   snapName: string;
 };
 
 export const SnapReport: FunctionComponent<SnapReportProps> = ({
+  snapId,
   snapName,
 }) => {
   const [showModal, setShowModal] = useState(false);
+  // TODO  need to changed in future when we store the reported snaps in redux store.
   const [reported, setReported] = useState(false);
 
-  const onSign = async () => {
-    setShowModal(false);
-    setReported(true);
+  const { address } = useAccount();
+  const { showSuccessMsg } = useToastMsg();
+
+  const { signMessage, signError, snapVCBuilder } = useVerifiableCredential();
+
+  useSignErrorHandler(signError);
+
+  // TODO: hardcode options for now, change to dynamic if needed,
+  // TODO may need to consider change options to key value pair to support i18n, key to stored in DB, value is localized string shown in UI.
+  const options = [t`Scam`, t`Vulnerable`];
```

Comment:
i think it is okay for now when the options is not confirmed

and i would say we can create another PR to modify the multicheckbox with options that allow use to provide value and label

Reply:
hi, @ritave as what @stanleyyconsensys said, lets address the key value pairs when we change the multicheckbox component in future PR.